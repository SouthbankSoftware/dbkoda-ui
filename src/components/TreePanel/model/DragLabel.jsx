/**
* @Author: Wahaj Shamim <wahaj>
* @Date:   2017-03-15T10:54:51+11:00
* @Email:  wahaj@southbanksoftware.com
* @Last modified by:   wahaj
* @Last modified time: 2017-03-17T09:23:03+11:00
*/

import React from 'react';
import { DragSource } from 'react-dnd';
import { DragItemTypes } from '#/common/Constants.js';

const labelSource = {
  beginDrag(props) {
    return { label: props.label };
  },
};

function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging(),
  };
}

/**
 * Defines the label of tree component which is dragable using react-dnd
 */
class DragLabel extends React.Component {
  static get defaultProps() {
    return {
      filter: ''
    };
  }
  get FilteredTextLabel() {
    const filterText = this.props.filter;
    const strText = this.props.label;
    const matchStart = strText.toLowerCase().indexOf(filterText.toLowerCase());
    if (filterText != '' && matchStart >= 0) {
      const matchEnd = matchStart + (filterText.length - 1);
      const beforeMatch = strText.slice(0, matchStart);
      const matchText = strText.slice(matchStart, matchEnd + 1);
      const afterMatch = strText.slice(matchEnd + 1);
      return <span>{beforeMatch}<mark>{matchText}</mark>{afterMatch}</span>;
    }
    return strText;
  }
  render() {
    const { connectDragSource, isDragging } = this.props;
    return connectDragSource(
      <span
        style={{
          opacity: isDragging ? 0.5 : 1,
          cursor: 'move',
        }}
      >{this.FilteredTextLabel}</span>,
    );
  }
}

DragLabel.propTypes = {
  label: React.PropTypes.string.isRequired,
  filter: React.PropTypes.string.isRequired,
  connectDragSource: React.PropTypes.func.isRequired,
  isDragging: React.PropTypes.bool.isRequired,
};

export default DragSource(DragItemTypes.LABEL, labelSource, collect)(DragLabel);
