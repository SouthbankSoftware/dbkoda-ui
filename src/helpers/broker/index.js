import {Broker} from './Broker';
import EventType from './EventType';

export default {Broker: Broker, EventType: EventType};
