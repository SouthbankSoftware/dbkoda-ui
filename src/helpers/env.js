/**
 * @Last modified by:   guiguan
 * @Last modified time: 2018-07-04T16:38:07+10:00
 *
 * dbKoda - a modern, open source code editor, for MongoDB.
 * Copyright (C) 2017-2018 Southbank Software
 *
 * This file is part of dbKoda.
 *
 * dbKoda is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * dbKoda is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with dbKoda.  If not, see <http://www.gnu.org/licenses/>.
 */

import _ from 'lodash';

// default logging API
global.l = {
  error: console.error,
  warn: console.warn,
  notice: console.log.bind(null, '%cNotice: ', 'color: green'),
  info: console.log,
  log: console.log,
  debug: console.log.bind(null, '%cDebug: ', 'color: blue')
};

// defaults for webpack
global.VERSION = '';
global.IS_STORYBOOK = false;

global.IS_PRODUCTION = process.env.NODE_ENV === 'production';
global.IS_DEVELOPMENT = !IS_PRODUCTION;
global.IS_TEST = process.env.NODE_ENV === 'test';
global.IS_ELECTRON = _.has(window, 'process.versions.electron');
global.WINDOW_NAME = window.location.pathname === '/ui/' ? 'main' : 'performance';
global.IS_FEATHERS_READY = false;

if (IS_ELECTRON) {
  // fix the problem that process.platform is not available at start
  process.platform = window.require('os').platform();

  const { remote } = window.require('electron');

  global.UAT = remote.getGlobal('UAT');
  global.PATHS = remote.getGlobal('PATHS');
} else {
  global.UAT = false;

  if (!IS_TEST) {
    // eslint-disable-next-line no-undef
    global.PATHS = WEBPACK_PATHS;
  } else {
    global.PATHS = {};
  }
}

global.sendToMain = (channel, ...args) => {
  if (IS_ELECTRON) {
    const { ipcRenderer } = window.require('electron');

    ipcRenderer.send(channel, ...args);
  }
};

global.logToMain = (level, message, options) => global.sendToMain('log', level, message, options);

export const protocol = 'http://';
export const host = '127.0.0.1';

const _port = IS_ELECTRON && window.require('electron').remote.getGlobal('CONTROLLER_PORT');
export const port = typeof _port === 'number' ? _port : 3030;
export const url = protocol + host + ':' + port;
export const analytics = {
  development: 'UA-101162043-2',
  prod: 'UA-101162043-1'
};
