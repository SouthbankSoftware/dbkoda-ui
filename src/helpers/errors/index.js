/*
 * dbKoda - a modern, open source code editor, for MongoDB.
 * Copyright (C) 2017-2018 Southbank Software
 *
 * This file is part of dbKoda.
 *
 * dbKoda is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * dbKoda is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with dbKoda.  If not, see <http://www.gnu.org/licenses/>.
 */
/**
 * Created by joey on 30/6/17.
 */

export const ErrorCodes = {
  BAD_VALUE: 2,
  UNKNOWN_ERROR: 8,
  NAMESPACE_NOT_FOUND: 26,
  EXCEEDED_TIME_LIMIT: 50,
  COMMAND_NOT_FOUND: 59,
  WRITE_CONCERN_ERROR: 64,
  NOT_MASTER: 10107,
  DUPLICATE_KEY: 11000,
  NOT_MASTER_NO_SLAVE_OK: 13435,
  NOT_MASTER_OR_SECONDARY: 13436,
  CANT_OPEN_DB_IN_READ_LOCK: 15927,
};
