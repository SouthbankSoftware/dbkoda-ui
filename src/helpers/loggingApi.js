/**
 * @flow
 *
 * @Author: Guan Gui <guiguan>
 * @Date:   2018-05-04T10:41:36+10:00
 * @Email:  root@guiguan.net
 * @Last modified by:   guiguan
 * @Last modified time: 2018-05-08T14:58:32+10:00
 *
 * dbKoda - a modern, open source code editor, for MongoDB.
 * Copyright (C) 2017-2018 Southbank Software
 *
 * This file is part of dbKoda.
 *
 * dbKoda is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * dbKoda is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with dbKoda.  If not, see <http://www.gnu.org/licenses/>.
 */

import _ from 'lodash';
import util from 'util';

global.rg4js = null;
export let isRaygunEnabled = false; // eslint-disable-line
let cachedUserObj;

export const setUser = (user: { id: string }) => {
  const userObj = {
    identifier: user.id
  };

  if (global.rg4js) {
    global.rg4js('setUser', userObj);
  } else {
    cachedUserObj = userObj;
  }
};

export const sendError = (error: Error, options: { customData?: *, tags?: * }) => {
  const { customData, tags } = options || {};

  global.rg4js &&
    global.rg4js('send', {
      error,
      customData,
      tags
    });
};

let defaultTags;
let defaultPiggybackTags;

const initRaygun = (manualBoot = true) => {
  global.rg4js = require('raygun4js');

  // NOTE: this is not a global variable but a placeholder string that will be replaced by webpack
  // DefinePlugin. The API_KEY is retrieved automatically from package.json at the building time of
  // ui bundle
  // $FlowFixMe
  global.rg4js('apiKey', API_KEY); // eslint-disable-line no-undef
  global.rg4js('enablePulse', true);
  // NOTE: this is not a global variable but a placeholder string that will be replaced by webpack
  // DefinePlugin. The VERSION is retrieved automatically from package.json at the building time of
  // ui bundle
  // $FlowFixMe
  global.rg4js('setVersion', VERSION); // eslint-disable-line no-undef
  global.rg4js('saveIfOffline', true);

  if (cachedUserObj) {
    global.rg4js('setUser', cachedUserObj);
    cachedUserObj = null;
  } else {
    setUser({ id: 'beforeConfigLoaded' });
  }

  global.rg4js('withTags', defaultTags);

  global.rg4js('onBeforeSend', payload => {
    if (isRaygunEnabled) {
      return payload;
    }

    return false;
  });

  global.rg4js('onBeforeSendRUM', payload => {
    if (isRaygunEnabled) {
      return payload;
    }

    return false;
  });

  manualBoot && global.rg4js('boot');
};

export const infoSymbol = Symbol.for('info');

const attachGlobalErrorEvents = () => {
  window.onerror = (message, _source, _lineno, _colno, error) => {
    l.error(error || message, {
      // $FlowFixMe
      [infoSymbol]: {
        tags: ['unhandled-exception']
      }
    });

    return true;
  };

  window.onunhandledrejection = event => {
    event.preventDefault();

    l.error(event.reason, {
      // $FlowFixMe
      [infoSymbol]: {
        tags: ['unhandled-rejection']
      }
    });
  };
};

export const toggleRaygun = (enabled: boolean, manualBoot: boolean = true) => {
  if (enabled) {
    isRaygunEnabled = true;

    // lazy init raygun
    if (!global.rg4js) {
      initRaygun(manualBoot);
    }
  } else {
    isRaygunEnabled = false;
  }
};

export const initLoggingApi = (tags: string[]) => {
  defaultPiggybackTags = tags.slice();
  defaultPiggybackTags.push('piggyback');

  tags.push(global.IS_PRODUCTION ? 'prod' : 'dev');
  global.UAT && tags.push('uat');
  defaultTags = tags;

  // $FlowFixMe
  console._error = console.error;

  global.logger = global.l = {
    error: null,
    warn: null,
    notice: null,
    info: null,
    debug: null
  };

  _.forEach(l, (v, k) => {
    if (v) return;

    l[k] = (...args) => {
      if (k === 'error') {
        const results = [];
        let resultsWithErrors;
        const info = {};

        for (const arg of args) {
          const argType = typeof arg;
          let isIgnored = false;

          if (argType === 'object' && arg) {
            const c = arg[infoSymbol];

            if (c) {
              isIgnored = true;
              _.assign(info, c);
            }

            if (info.raygun !== false && arg instanceof Error) {
              isIgnored = true;

              if (!resultsWithErrors) {
                // lazy init
                resultsWithErrors = results.slice();
              }

              // shallow clone original error
              info.error = Object.create(
                Object.getPrototypeOf(arg),
                // $FlowFixMe
                Object.getOwnPropertyDescriptors(arg)
              );

              results.push(arg);
              resultsWithErrors.push(arg.message);
            }
          }

          if (!isIgnored) {
            results.push(arg);
            resultsWithErrors && resultsWithErrors.push(arg);
          }
        }

        // stringify input args
        // $FlowFixMe
        info.message = util.format(...results);

        if (info.error) {
          // $FlowFixMe
          info.error.message = util.format(...resultsWithErrors);
          info.error.stack = info.error.stack.replace(
            /^.*/,
            `${info.error.constructor.name}: ${info.error.message}`
          );
        } else if (info.raygun !== false) {
          info.error = new Error(info.message);
        }

        // $FlowFixMe
        console._error(info.error);

        const usePiggyback = !isRaygunEnabled && info.raygun !== false;

        if (usePiggyback) {
          if (info.tags) {
            info.tags = [...defaultPiggybackTags, ...info.tags];
          } else {
            info.tags = defaultPiggybackTags;
          }
        }

        // piggyback error via main (dbkoda)
        logToMain('error', info.error.message, {
          usePiggyback,
          stack: info.error.stack,
          info: usePiggyback ? _.omit(info, ['message', 'error']) : undefined
        });

        if (isRaygunEnabled && info.raygun !== false) {
          sendError(info.error, info);
        }
      } else if (k === 'warn') {
        console.warn(...args);

        logToMain('warn', util.format(...args));
      } else if (k === 'notice') {
        console.log('%cNotice: ', 'color: green', ...args);

        logToMain('notice', util.format(...args));
      } else if (k === 'info') {
        console.log(...args);
      } else if (k === 'debug') {
        if (!IS_PRODUCTION) {
          console.log('%cDebug: ', 'color: blue', ...args);
        }
      }
    };
  });

  l._error = (...args) => {
    // $FlowFixMe
    l.error(...args, { [infoSymbol]: { raygun: false } });
  };

  // $FlowFixMe
  console.error = l.error;

  attachGlobalErrorEvents();
};
