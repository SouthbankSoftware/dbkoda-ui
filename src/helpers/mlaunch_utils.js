/**
 * @Author: guiguan
 * @Date:   2017-03-31T09:12:27+11:00
 * @Last modified by:   guiguan
 * @Last modified time: 2017-04-18T11:23:34+10:00
 */

const shelljs = require('shelljs');

const MLAUNCH = 'mlaunch';
const TMP_DIR = 'data';

/**
 * generate random port number between 6000 and 7000
 * @returns {number}
 */
const getRandomPort = () => {
  return Math.floor(Math.random() * 7000) + 6000;
};

/**
 * launch mongodb instance
 *
 * @param  type [description]
 */
const launchMongoInstance = (type, port, parameters) => {
  const command = MLAUNCH +
    ' init ' +
    type +
    ' --dir ' +
    TMP_DIR +
    '/' +
    port +
    ' --port ' +
    port +
    ' ' +
    parameters;
  console.log('launch command ', command);
  shelljs.exec(command);
};

/**
 * launch a single mongodb instance
 *
 */
const launchSingleInstance = (port, parameters = '') => {
  launchMongoInstance('--single', port, parameters);
};

/**
 * launch replica set
 *
 */
const launchReplicaSet = (port, nodenumber, parameters) => {
  console.log('launch replica set ', port);
  launchMongoInstance('--replicaset', port, '--nodes ' + nodenumber + ' ' + parameters);
};

/**
 * launch a mongos clusters
 * @param port
 * @param nodenumber
 * @param parameters
 */
const launchMongos = (port, nodenumber, parameters) => {
  launchMongoInstance('--mongos ' + nodenumber, port, parameters);
};

/**
 * generate mongo dump data on the collection
 *
 * @param port
 * @param dbName
 * @param colName
 * @param parameters
 */
const generateMongoData = (port, dbName = 'test', colName = 'test', parameters = '') => {
  const command = 'mgenerate tests/integration/json_template.json --num 1 --port ' +
    port +
    ' --database ' +
    dbName +
    ' --collection ' +
    colName +
    ' ' +
    parameters;
  shelljs.exec(command);
};

/**
 * shutdown mongo instance based on port number
 *
 * @param port
 */
const killMongoInstance = (port) => {
  const command = MLAUNCH + ' kill --dir ' + TMP_DIR + '/' + port;
  shelljs.exec(command);
  shelljs.exec('rm -fr ' + TMP_DIR + '/' + port);
};

module.exports = {
  launchSingleInstance,
  killMongoInstance,
  launchMongoInstance,
  launchReplicaSet,
  launchMongos,
  getRandomPort,
  generateMongoData,
};
