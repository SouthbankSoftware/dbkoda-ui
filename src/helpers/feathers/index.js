import load from 'little-loader';
import feathers from 'feathers-client';
import _ from 'lodash';
import {url} from '../../env';

let instance = false;


/**
 * featherjs client wrapper. It wraps all featherjs client calls for each component.
 */
class FeatherClient {

  constructor() {
    this.feathers = feathers().configure(feathers.hooks());
    this.outputListeners = [];
  }

  configurePrimus(primus) {
    this.feathers = this.feathers.configure(feathers.primus(primus));
    this.shellService = this.feathers.service('/mongo-shells');
    const that = this;
    this.shellService.on('shell-output', (output) => {
      const {id, shellId} = output;
      const listeners = that.getShellOutputListeners(parseInt(id, 10), parseInt(shellId, 10));
      for (const ls of listeners) {
        for (const l of ls.listeners) {
          l(output);
        }
      }
    });
  }

  service(service) {
    return this.feathers.service(service);
  }

  /**
   * add listener on the shell output via websocket
   *
   * @param connectionId  the mongodb connection id
   * @param shellId the shell connection id
   * @param listener  the listener who is listening on that connection
   */
  addOutputListener(connectionId, shellId, listener) {
    const output = this.getShellOutputListeners(connectionId, shellId);
    if (output.length === 0) {
      this.outputListeners.push({connectionId, shellId, listeners: [listener]});
    } else {
      output[0].listeners.push(listener);
    }
  }

  /**
   * remove output listener from featherjs client
   *
   * @param connectionId
   * @param shellId
   * @param listener
   */
  removeOutputListener(connectionId, shellId, listener) {
    const output = this.getShellOutputListeners(connectionId, shellId);
    if (output.length > 0) {
      const idx = output[0].listeners.indexOf(listener);
      if (idx >= 0) {
        output[0].listeners.splice(idx, 1);
      }
    }
  }

  getShellOutputListeners(connectionId, shellId) {
    return _.filter(this.outputListeners, {connectionId, shellId});
  }

}

export const featherClient = () => {
  if (instance) return instance;
  instance = new FeatherClient();
  instance.configurePrimus(new Primus(url));
  return instance;
};

// let times = 0;
// const loadPrimus = () => {
//   load(url + '/dist/primus.js', (err) => {
//     if (!err) {
//       const primus = new Primus(url);
//       featherClient().configurePrimus(primus);
//     } else {
//       times += 1;
//       if (times < 3) {
//         setTimeout(() => {
//           loadPrimus();
//         }, 3000);
//       }
//     }
//   });
// };
// if (process.env.NODE_ENV !== 'test') {
//   loadPrimus();
// } else {
//
// }
