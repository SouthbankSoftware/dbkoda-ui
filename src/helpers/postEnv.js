/**
 * @Author: Guan Gui <guiguan>
 * @Date:   2018-07-03T16:53:55+10:00
 * @Email:  root@guiguan.net
 * @Last modified by:   guiguan
 * @Last modified time: 2018-07-03T17:00:37+10:00
 *
 * dbKoda - a modern, open source code editor, for MongoDB.
 * Copyright (C) 2017-2018 Southbank Software
 *
 * This file is part of dbKoda.
 *
 * dbKoda is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * dbKoda is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with dbKoda.  If not, see <http://www.gnu.org/licenses/>.
 */

import * as d3 from 'd3';
import { initLoggingApi } from '~/helpers/loggingApi';

// because of buggy globalize, we have to separate this out
if (!IS_TEST) {
  initLoggingApi(['dbkoda-ui', `window-${global.WINDOW_NAME}`]);

  const Globalize = require('globalize'); // doesn't work well with import

  global.Globalize = Globalize;

  const { language, region } = Globalize.locale().attributes;

  global.locale = `${language}-${region}`;

  d3.formatDefaultLocale(require(`d3-format/locale/${locale}.json`)); // eslint-disable-line import/no-dynamic-require

  global.globalString = (path, ...params) => Globalize.messageFormatter(path)(...params);
  global.globalNumber = (value, config) => Globalize.numberFormatter(config)(value);
}
