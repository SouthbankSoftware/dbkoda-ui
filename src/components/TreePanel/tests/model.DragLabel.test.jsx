/**
 * @Author: Wahaj Shamim <wahaj>
 * @Date:   2017-03-22T17:09:32+11:00
 * @Email:  wahaj@southbanksoftware.com
 * @Last modified by:   guiguan
 * @Last modified time: 2017-11-21T11:00:07+11:00
 *
 * dbKoda - a modern, open source code editor, for MongoDB.
 * Copyright (C) 2017-2018 Southbank Software
 *
 * This file is part of dbKoda.
 *
 * dbKoda is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * dbKoda is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with dbKoda.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import '~/helpers/configEnzyme';
import { mount } from 'enzyme';
import { useStrict } from 'mobx';
import { DragDropContext } from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import renderer from 'react-test-renderer';

import DragLabel from '#/TreePanel/model/DragLabel';

describe('DragLabel', () => {
  const props = {
    id: 'id.location.type.host.com:8888',
    label: 'long.text.label.with.south:8888',
    type: 'shard'
  };
  const props1 = {
    id: 'root_databases_DBEnvyLoad',
    label: 'DBEnvyLoad',
    type: 'database'
  };

  const treeState = {
    filter: '',
    setFilter: (value) => {
      treeState.filter = value;
    }
  };

  class DragLabelTest extends React.Component {
    render() {
      return (
        <DragLabel id={this.props.id} label={this.props.label} type={this.props.type} treeState={this.props.treeState} />
      );
    }
  }

  const DDCDragLabel = DragDropContext(HTML5Backend)(DragLabelTest);

  beforeAll(() => {
    useStrict(true);
  });

  test('check label text', () => {
    const label = mount(
      <DDCDragLabel id={props.id} label={props.label} type={props.type} treeState={treeState} />,
    );
    expect(label.find('span').text()).toEqual('long:8888');
  });

  test('matches snapshot', () => {
    const testLabel = renderer.create(
      <DDCDragLabel id={props.id} label={props.label} type={props.type} treeState={treeState} />,
    );

    expect(testLabel).toMatchSnapshot();
  });

  test('check label text with filter', () => {
    treeState.setFilter('south');
    const label = mount(
      <DDCDragLabel id={props.id} label={props.label} type={props.type} treeState={treeState} />,
    );

    expect(label.find('mark').text()).toEqual(
      treeState.filter,
    );
  });

  test('check label text with filter', () => {
    const label1 = mount(
      <DDCDragLabel id={props1.id} label={props1.label} type={props1.type} treeState={treeState} />,
    );

    expect(label1.find('span').text()).toEqual(props1.label);
  });
});
