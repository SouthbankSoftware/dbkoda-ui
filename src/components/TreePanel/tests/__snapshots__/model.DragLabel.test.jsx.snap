// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragLabel matches snapshot 1`] = `
<span
  style={
    Object {
      "cursor": "move",
      "opacity": 1,
    }
  }
>
  long:8888
</span>
`;
